src/realm/object-store/sync/app.hpp

```diff
@@ -307,6 +326,25 @@ class App : public std::enable_shared_from_this<App>,
     link_user(const std::shared_ptr<SyncUser>& user, const AppCredentials& credentials,
               util::UniqueFunction<void(const std::shared_ptr<SyncUser>&, util::Optional<AppError>)>&& completion);
 
+    /// Refreshes the user credentials without logging the user out locally. This function will use
+    /// the provided credentials to log in again on the server, resulting in new refresh and access
+    /// tokens and the latest profile information from the server will also be retrieved. This cannot
+    /// be used to link the user with a new identity, nor can it be used with Anonymous credentials.
+    /// The user will be logged in if currently logged out and a new user object will be provided to
+    /// the callback.
+    /// On success the user will be returned with updated auth tokens information.
+    /// NOTE: This function is primarily for re-logging in the user without closing the realms when
+    /// switching between cloud and edge servers.
+    ///
+    /// @param user The user which will have the credentials refreshed, the user must be logged in
+    /// @param credentials The `AppCredentials` used to log the user in again.
+    /// @param completion The completion handler to call when the log in operation is complete.
+    ///                   If the operation is  successful, the result will contain the original
+    ///                   `SyncUser` object representing the user.
+    void
+    refresh_user(const std::shared_ptr<SyncUser>& user, const AppCredentials& credentials,
```

Comment:
what happens with the sync session while the user re-logs in?

Reply:
Sorry, I thought I removed this function. Looks like I forgot to remove it from the header.